src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -873,6 +876,34 @@ internal bool CanPaste(object parameters)
 
         private void Paste(object parameter)
         {
+            var locatableModels = model.ClipBoard.Where(item => item is NoteModel || item is NodeModel);
+
+            bool outOfView = locatableModels.Any(item => !model.CurrentWorkspace.Rect.Contains(item.Rect));
+
+            // If copied nodes are out of view, we paste their copies under mouse cursor or at the center of workspace.
+            if (outOfView)
+            {
+                var dynamoWindow = parameter as DynamoView;
```

Comment:
I and @marimano had some issues around this code. There is no easy solution how to get mouse position, during Ctrl+V. What do you think @ikeough ?